### PR TITLE
Remove initialize functions from the snowball package

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

Down with initialize functions.

## How this works

CTRL + X

## How this was tested

Existing UT

### CODE REVIEW

File: snow/consensus/snowball/flat.go [hunk 1]
```diff
@@ -11,11 +11,10 @@ import (
 var _ Consensus = (*Flat)(nil)
 
 func NewFlat(params Parameters, choice ids.ID) Consensus {
-	f := &Flat{
-		params: params,
+	return &Flat{
+		params:        params,
+		nnarySnowball: newNnarySnowball(params.BetaVirtuous, params.BetaRogue, choice),
```
Comment (StephenButtolph) [inline snow/consensus/snowball/flat.go:16]: Super nitty - but I try to initialize in the order that the struct defines
```suggestion
		nnarySnowball: newNnarySnowball(params.BetaVirtuous, params.BetaRogue, choice),
		params:        params,
```

File: snow/consensus/snowball/nnary_snowball.go [hunk 2]
```diff
@@ -11,6 +11,14 @@ import (
 
 var _ NnarySnowball = (*nnarySnowball)(nil)
 
+func newNnarySnowball(betaVirtuous, betaRogue int, choice ids.ID) nnarySnowball {
+	return nnarySnowball{
+		preference:         choice,
+		numSuccessfulPolls: make(map[ids.ID]int),
+		nnarySnowflake:     newNnarySnowflake(betaVirtuous, betaRogue, choice),
```
Comment (StephenButtolph) [inline snow/consensus/snowball/nnary_snowball.go:18]:
```suggestion
		nnarySnowflake:     newNnarySnowflake(betaVirtuous, betaRogue, choice),
		preference:         choice,
		numSuccessfulPolls: make(map[ids.ID]int),
```

File: snow/consensus/snowball/nnary_snowflake.go [hunk 3]
```diff
@@ -11,6 +11,14 @@ import (
 
 var _ NnarySnowflake = (*nnarySnowflake)(nil)
 
+func newNnarySnowflake(betaVirtuous, betaRogue int, choice ids.ID) nnarySnowflake {
+	return nnarySnowflake{
+		betaVirtuous: betaVirtuous,
+		betaRogue:    betaRogue,
+		nnarySlush:   newNnarySlush(choice),
```
Comment (StephenButtolph) [inline snow/consensus/snowball/nnary_snowflake.go:18]:
```suggestion
		nnarySlush:   newNnarySlush(choice),
		betaVirtuous: betaVirtuous,
		betaRogue:    betaRogue,
```

**Review by StephenButtolph — APPROVED** [review]

**Review by StephenButtolph — APPROVED** [review]

File: snow/consensus/snowball/nnary_slush.go [hunk 4]
```diff
@@ -11,6 +11,12 @@ import (
 
 var _ NnarySlush = (*nnarySlush)(nil)
 
+func newNnarySlush(choice ids.ID) nnarySlush {
```
Comment (joshua-kim) [inline snow/consensus/snowball/nnary_slush.go:14]: I feel like we dont need this function and we can just use the struct directly

File: snow/consensus/snowball/binary_slush.go [hunk 5]
```diff
@@ -7,6 +7,12 @@ import "fmt"
 
 var _ BinarySlush = (*binarySlush)(nil)
 
+func newBinarySlush(choice int) binarySlush {
+	return binarySlush{
```
Comment (joshua-kim) [inline snow/consensus/snowball/binary_slush.go:11]: I feel like we dont need this function and we can use the struct directly

**Review by joshua-kim — COMMENTED** [review]
Left comments on the two functions we dont need, I think a few of them are on the edge so I'll leave them to your discretion on if you want to keep them or not

**Review by abi87 — APPROVED** [review]

**Review by dhrubabasu — APPROVED** [review]

File: snow/consensus/snowball/nnary_slush.go (hunk 4 above)
Comment (StephenButtolph) [reply snow/consensus/snowball/nnary_slush.go:14]: I feel like I kinda want to keep them just for consistency... In general all the `Nnary` stuff could be moved to a test file...

**Review by gyuho — APPROVED** [review]

File: snow/consensus/snowball/binary_slush.go (hunk 5 above)
Comment (danlaine) [reply snow/consensus/snowball/binary_slush.go:11]: I don't feel super strongly but it fits with the pattern elsewhere so 🤷

**Pull request merged**